draco/analysis/flagging.py

```diff
@@ -957,30 +957,44 @@ class RFIMask(task.SingleTask):
         channel to be flagged.
     stack_ind : int
         Which stack to process to derive flags for the whole dataset.
-    destripe : bool, optional
-        Deprecated option to remove the striping.
     """
 
     sigma = config.Property(proptype=float, default=5.0)
     tv_fraction = config.Property(proptype=float, default=0.5)
     stack_ind = config.Property(proptype=int)
-    destripe = config.Property(proptype=bool, default=False)
 
-    def process(self, sstream):
+    @overload
+    def process(self, sstream: containers.SiderealStream) -> containers.SiderealRFIMask:
+        ...
+
+    @overload
+    def process(self, sstream: containers.TimeStream) -> containers.RFIMask:
+        ...
+
+    def process(
+        self, sstream: Union[containers.TimeStream, containers.SiderealStream]
```

Comment:
Not super familiar with the @overload decorator, but is the typing needed here if it's already given by @overload? Seems redundant

Reply:
I believe it is, or the function body isn't type checked as far as I understand. I was surprised by this as well, it's a little annoying. This may be type checker dependent but that's why mypy says.